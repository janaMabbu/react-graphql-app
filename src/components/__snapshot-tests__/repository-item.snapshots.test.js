/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-boost'
import RepositoryItem from '../repository-item'


 const client = new ApolloClient({
  uri: `snapshotclient`
})

const repositories =[
  {
      "id": "1",
      "name": "repo1",
      "url": "https://github.com/facebook/zstd",
      "descriptionHTML": "<div>snapshottestcases/div>",
      "viewerHasStarred": true,
      "stargazers": {
        "totalCount": 1,
        "__typename": "StargazerConnection"
      },
      "__typename": "Repository"
    },
    {
      "id": "2",
      "name": "repo2",
      "url": "https://github.com/facebook/zstd",
      "descriptionHTML": "<div>snapshottestcases 23/div>",
      "viewerHasStarred": true,
      "stargazers": {
        "totalCount": 1,
        "__typename": "StargazerConnection"
      },
      "__typename": "Repository"
    },
    {
      "id": "2",
      "name": "repo3",
      "url": "https://github.com/facebook/zstd",
      "descriptionHTML": "<div>snapshottestcases 45/div>",
      "viewerHasStarred": true,
      "stargazers": {
        "totalCount": 1,
        "__typename": "StargazerConnection"
      },
      "__typename": "Repository"
    }

]

describe(`Repository-tem:snapshots`, () => {
      repositories.forEach(repo => {
        test(`testing Repository-item-snapshots`, () => {
          const tree = renderer.create(
            <ApolloProvider client ={client}>
            <RepositoryItem { ...repo } />
            </ApolloProvider>
          ).toJSON()
          expect(tree).toMatchSnapshot()
        })
      })
})