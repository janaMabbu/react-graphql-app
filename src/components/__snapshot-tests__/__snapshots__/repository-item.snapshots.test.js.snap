// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository-tem:snapshots testing Repository-item-snapshots 1`] = `
<div>
  <div
    className="css-yduxpk"
  >
    <h2>
      <a
        href="https://github.com/facebook/zstd"
        rel="noopener noreferrer"
        target="_blank"
      >
        repo1
      </a>
    </h2>
    <div>
      <button
        className="css-1oih4z3"
        onClick={[Function]}
        type="button"
      >
        1
         Unstar
      </button>
    </div>
  </div>
  <div
    className="css-1mkpf1j"
  >
    <div
      className="css-e2fs5t"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>snapshottestcases/div>",
        }
      }
    />
  </div>
</div>
`;

exports[`Repository-tem:snapshots testing Repository-item-snapshots 2`] = `
<div>
  <div
    className="css-yduxpk"
  >
    <h2>
      <a
        href="https://github.com/facebook/zstd"
        rel="noopener noreferrer"
        target="_blank"
      >
        repo2
      </a>
    </h2>
    <div>
      <button
        className="css-1oih4z3"
        onClick={[Function]}
        type="button"
      >
        1
         Unstar
      </button>
    </div>
  </div>
  <div
    className="css-1mkpf1j"
  >
    <div
      className="css-e2fs5t"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>snapshottestcases 23/div>",
        }
      }
    />
  </div>
</div>
`;

exports[`Repository-tem:snapshots testing Repository-item-snapshots 3`] = `
<div>
  <div
    className="css-yduxpk"
  >
    <h2>
      <a
        href="https://github.com/facebook/zstd"
        rel="noopener noreferrer"
        target="_blank"
      >
        repo3
      </a>
    </h2>
    <div>
      <button
        className="css-1oih4z3"
        onClick={[Function]}
        type="button"
      >
        1
         Unstar
      </button>
    </div>
  </div>
  <div
    className="css-1mkpf1j"
  >
    <div
      className="css-e2fs5t"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>snapshottestcases 45/div>",
        }
      }
    />
  </div>
</div>
`;
